### PR TITLE
app as a plugin

### PR DESCRIPTION
**Pull request by Aniket-Engg**

### CODE REVIEW

File: apps/remix-ide/src/app.js [hunk 1]
```diff
@@ -50,8 +51,16 @@ const Editor = require('./app/editor/editor')
...
 
-class AppComponent {
+const profile = {
+  name: 'app',
+  displayName: 'App',
+  description: 'Application',
+  methods: []
+}
+
+class AppComponent extends Plugin {
```
Comment (yann300) [inline apps/remix-ide/src/app.js:61]: cool, this need to be added to the engine and activated too.
cc @bunsenstraat cause that might interfer with what he's doing currently

File: apps/remix-ide/src/app/tabs/compile-tab.js [hunk 2]
```diff
@@ -145,16 +145,11 @@ class CompileTab extends CompilerApiMixin(ViewPlugin) { // implements ICompilerA
   }
 
   getAppParameter (name) {
-    // first look in the URL params then in the local storage
-    const params = this.queryParams.get()
-    const param = params[name] ? params[name] : this.config.get(name)
-    if (param === 'true') return true
-    if (param === 'false') return false
-    return param
+    this.call('app', 'getAppParameter', name)
```
Comment (yann300) [inline apps/remix-ide/src/app/tabs/compile-tab.js:148]: you forgot to `return` and  `await`

File: apps/remix-ide/src/app.js [hunk 3]
```diff
@@ -159,6 +185,7 @@ class AppComponent {
     const contextualListener = new ContextualListener({ editor })
 
     self.engine.register([
```
Comment (yann300) [inline apps/remix-ide/src/app.js:187]: you also need to to 
`await self.appManager.activatePlugin(['app'])` (line 293)

File: apps/remix-ide/src/app/tabs/compile-tab.js (hunk 2 above)
Comment (Aniket-Engg) [reply apps/remix-ide/src/app/tabs/compile-tab.js]: done

File: apps/remix-ide/src/app.js (hunk 3 above)
Comment (Aniket-Engg) [reply apps/remix-ide/src/app.js:187]: doing this doesn't remix and shows errors in browser console

File: apps/remix-ide/src/app/tabs/compile-tab.js [hunk 4]
```diff
@@ -145,16 +145,11 @@ class CompileTab extends CompilerApiMixin(ViewPlugin) { // implements ICompilerA
   }
 
   getAppParameter (name) {
```
Comment (yann300) [inline apps/remix-ide/src/app/tabs/compile-tab.js:147]: I think you still miss the await here

File: apps/remix-ide/src/app/tabs/compile-tab.js [hunk 5]
```diff
@@ -145,16 +145,11 @@ class CompileTab extends CompilerApiMixin(ViewPlugin) { // implements ICompilerA
...
-    // first look in the URL params then in the local storage
-    const params = this.queryParams.get()
-    const param = params[name] ? params[name] : this.config.get(name)
-    if (param === 'true') return true
-    if (param === 'false') return false
-    return param
+    return this.call('app', 'getAppParameter', name)
```
Comment (yann300) [inline apps/remix-ide/src/app/tabs/compile-tab.js:148]: you still miss the await here

**Pull request closed without merging**